### PR TITLE
Remove TurboBoost and various other Projectile cleanups

### PR DESCRIPTION
**Pull request by reaperrr**
Commits should be self-explanatory.

Note: All yaml changes except the RA StingerAA weapon have been produced by the provided upgrade rules, so they've been tested and work.

### CODE REVIEW

**Comment by reaperrr** [comment]
Updated, added fix for variable angle/speed on bullets. The old if statement would require BOTH to define two values to work, now they're independent from each other (and yes, I tested this, works just fine).

**Comment by ScottNZ** [comment]
Deprecate doesn't mean 'actually do the removal now'. The commit message could be a bit clearer.

**Comment by penev92** [comment]
Which of the 8 commit messages?

**Comment by ScottNZ** [comment]
The one that talks about deprecation

**Comment by reaperrr** [comment]
Silly me for assuming what it meant and not actually checking.

**Comment by Mailaender** [comment]
Already needs a rebase.

**Comment by reaperrr** [comment]
Rebased.

**Comment by Mailaender** [comment]
:+1: Those cleanups look good. I tested the bullet blocking and parabombs in the RA mod. Saw no regressions. :white_check_mark:

**Comment by reaperrr** [comment]
If any of you spot something that could be optimised a little further (mostly with regards to style and performance optimisations, i.e. commit 7 / ff98750), don't hesitate to point it out. 
There were a few cases where I wasn't sure whether making something readonly might have unwanted side-effects, as well as some cases where I wasn't sure whether moving something from tick to constructor - a calculation, for example - would have any benefit for performance or might break something, so I left them as they are.

Just saying.

**Comment by Mailaender** [comment]
> There were a few cases where I wasn't sure whether making something readonly might have unwanted side-effects

The compiler would catch that and I also don't believe the performance improvements are that great as the compiler already knows. It probably places those automatically if we forget it. The annotation may be purely for humans to keep an overview.

> as well as some cases where I wasn't sure whether moving something from tick to constructor - a calculation, for example - would have any benefit for performance or might break something, so I left them as they are.

```csharp
acceleration = new WVec(WRange.Zero, WRange.Zero, info.Acceleration);
```
is not really a calculation, but conversion that only consists of constants loaded at map loading so it will be safe. Performance improvements won't be too huge, but still this is a good cleanup.

**Comment by Mailaender** [comment]
This pull request is already touching enough so I would recommend to stop here for now.

File: mods/cnc/weapons/missiles.yaml [hunk 1]
```diff
@@ -6,10 +6,10 @@ Rockets:
 	ValidTargets: Ground, Air
 	Projectile: Missile
 		Arm: 0
-		High: yes
+		Blockable: false
```
Comment (Phrohdoh) [inline mods/cnc/weapons/missiles.yaml]: Why are these being changed from true to false?

**Comment by reaperrr** [comment]
Rebased.

**Comment by pchote** [comment]
Code changes look good, but haven't tested ingame so :+0.5:

**Comment by reaperrr** [comment]
Rebased.

**Comment by penev92** [comment]
Changes look good, and things still shoot in the game, so :+1:

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/7a7d69845f20e5c40c9c1266101f7f04ea2b5d52...e68794ead25d6a9b7e6925f9560fe117c6725439)

**Pull request merged**